core/src/main/java/org/apache/accumulo/core/classloader/ClassLoaderUtil.java

```diff
@@ -75,22 +75,29 @@ static synchronized void resetContextFactoryForTests() {
 
   @SuppressWarnings("deprecation")
   public static ClassLoader getClassLoader(String context) {
-    try {
-      if (FACTORY.isValid(context)) {
-        return FACTORY.getClassLoader(context);
-      } else {
-        return org.apache.accumulo.start.classloader.vfs.AccumuloVFSClassLoader.getClassLoader();
-      }
-    } catch (RuntimeException e) {
+    if (context != null && !context.isEmpty()) {
+      return FACTORY.getClassLoader(context);
```

Comment:
IIRC @EdColeman had commented to me that only checking when the property is set could also lead to an issue if the context definition / configuration changes afterwards. That is why I'm also checking the validity of context name in getClassLoader in #3678 .

Reply:
Yes - one of the issues is that the way the context is being used, there have been "bad changes" where contexts that are in use have been deleted / removed.  So what was once valid when the context was set, it becomes invalid at some point.

It may be that we can only do so much to insure that a context is valid and remains valid for the lifetime of its usage.  If it was correct when set, then it may not be a big leap to assume that it can be kept that way.

However, protecting the system in the face of context errors so that it does not put things into an unrecoverable position should be a design goal.  The errors could be because of the user, or maybe a system / component issue - I think that is the path that the PR for preventing half-dead tservers moves in that direction and may be out of the scope of this PR?

Reply:
I don't think it possible to avoid bad config being set.  We could do complex validation of any classloader config set via ZK that exmanines the new config against all existing classloader config.  However since the config could be set in a file, its still possible to set bad config.  

Its way out of scope of this PR which is focused on validation of a single config, but maybe we could create something like the following for loading table plugins.  After creating the following utility method somewhere we could then adapt code that instantiates classes to use the new utility method. 

```java
class SomeUtilClass {
    /**
     * Creates a new instance of an object using a tables context classloader retrying until the object is created.
     * /
    T newInstance(TableConfiguration config, Property prop, Class<T> expectedType){
         // TODO use retry
         while(true){
              try{
                String context = config.getClassloaderContext();
                String clazz = config.getClassName(prop);
                return instantiateClass(context, clazz);
               } catch(Exception e){
                   // assume this is caused by bad classloader config and retry
                   // TODO log message and sleep using retry
              }
         }
     }
}
```

Reply:
I agree with `I don't think it possible to avoid bad config being set` - but I though these changes where an attempt to validate that, at a minimum an existing path was set?  

One complication (of many) is that the classpath maybe a regex, that may or may not contain the needed jars / classes - it is only when the use code tries to access a class that it is know to be present.  There is no way for our general code to know which jars / classes will be needed / used at any given point by the users.

Or have I lost the thread on this and misunderstanding the scope of the check and what we are attempting to protect against?

Reply:
The original intent was to validate that `table.class.loader.context` is a valid context name in the configuration. Valid, I think, means that the ClassLoaderContextFactory implementation to resolve the name and construct a ClassLoader for it. Validating the classpath for that context is beyond the scope.